iree_tests/benchmarks/benchmark_sdxl_rocm.py

```diff
@@ -1,36 +1,132 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
 import os
 from collections import namedtuple
+import logging
+from typing import Sequence
+import subprocess
 
 benchmark_dir = os.path.dirname(os.path.realpath(__file__))
+iree_root = os.path.dirname(benchmark_dir)
+prompt_encoder_dir = f"{iree_root}/pytorch/models/sdxl-prompt-encoder-tank"
+scheduled_unet_dir = f"{iree_root}/pytorch/models/sdxl-scheduled-unet-3-tank"
+vae_decode_dir = f"{iree_root}/pytorch/models/sdxl-vae-decode-tank"
 
-def launchCommand(scriptcommand):
-    print("Launching:", scriptcommand, "[ Proc:", os.getpid(), "]")
-    try:
-        ret = os.system(scriptcommand)
-        return ret
-    except OSError as errormsg:
-        print(
-            "Invoking ",
-            scriptcommand,
-            " failed:",
-            errormsg,
-        )
-        return 1
+def run_iree_command(args: Sequence[str] = ()):
+    command = "Exec:", " ".join(args)
+    logging.getLogger().info(command)
+    proc = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False)
+    stdout_v, stderr_v, = proc.stdout, proc.stderr
+    return_code = proc.returncode
+    if return_code == 0:
+        return 0, proc.stdout
+    logging.getLogger().info(f"Command failed with error: {proc.stderr}")
+    return 1, proc.stdout
 
 def run_sdxl_rocm_benchmark():
-    scriptcommand = (
-        "bash "
-        + f"{benchmark_dir}/benchmark_sdxl_rocm.sh"
-    )
-    return launchCommand(scriptcommand)
+    exec_args = [
+        "iree-compile",
+        f"{benchmark_dir}/sdxl_pipeline_bench_f16.mlir",
+        "--iree-hal-target-backends=rocm",
+        "--iree-rocm-target-chip=gfx90a",
+        "--iree-global-opt-propagate-transposes=true",
+        "--iree-codegen-llvmgpu-use-vector-distribution",
+        "--iree-codegen-gpu-native-math-precision=true",
+        "--iree-rocm-waves-per-eu=2",
```

Comment:
If flags get shared between benchmarks and tests, we could move them to a flagfile. As a general rule of thumb for this sort of infrastructure, we should only benchmark models that we also test. If we benchmark a model without testing it too, then we might accidently end up benchmarking a model that doesn't actually perform the computations we want it to.

Reply:
Got it, the yaml file is setup so that the benchmarking job only executes if the testing for the models pass

Reply:
Are the flags the same? They should be pulled from the exact same file to avoid drift

Reply:
We can create a flag file in future for sure. Not seeing any major flags being shared amongst the benchmarks at the moment

Reply:
Oh your comments didn't load for some reason. Yes, the flags are the same as the ones in the iree_tests/configs/ json files. It seems a bit messy to copy the process to convert and check json files that you did originally here too. These are just the input and ouput dimensions, so pretty rare that they will change. What do you think?

Reply:
What I had in mind was extracting these sorts of flags into a flagfile.txt: https://github.com/nod-ai/SHARK-TestSuite/blob/16812962da23c85302c03ef505f4c654cb54b46c/iree_tests/configs/config_sdxl_scheduled_unet_gpu_rocm.json#L8-L18

Then in that file using `--flagfile=${IREE_TEST_PATH_EXTENSION}/sdxl_rocm_compile_flags.txt` and similarly sourcing the file here too.

Juggling this many flags across configurations gets unwieldy no matter how we do it though, so we should be moving towards configuration in .mlir files or some other purpose-built API IMO. I guess for now let's keep the duplication and just be careful about letting the complexity spread too much.

Reply:
Ah got it. Yeah we can definitely do that